src/libraries/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -66,6 +66,72 @@ public static void BuildChain()
             }
         }
 
+        public enum BuildChainWithNotSignatureValidTest : int
+        {
+            TrustedRoot,
+            UntrustedRoot,
+            PartialChain
+        }
+
+        public static IEnumerable<object[]> BuildChainWithNotSignatureValidData()
+        {
+            yield return new object[] { true, X509ChainStatusFlags.NoError, BuildChainWithNotSignatureValidTest.TrustedRoot };
+            yield return new object[] { false, X509ChainStatusFlags.UntrustedRoot | X509ChainStatusFlags.NotSignatureValid, BuildChainWithNotSignatureValidTest.UntrustedRoot };
+            yield return new object[] { false, X509ChainStatusFlags.PartialChain, BuildChainWithNotSignatureValidTest.PartialChain };
+        }
+
+        [Theory]
+        [MemberData(nameof(BuildChainWithNotSignatureValidData))]
+        [PlatformSpecific(TestPlatforms.Linux)] // NotSignatureValid gets ignored on the root certificate.
+        public static void BuildChainWithNotSignatureValid(
```

Comment:
This test should move to DynamicChainTests.cs; just because that's where all of the tests that involve `TestDataGenerator.MakeTestChain3` live.  (This file uses static data, that one uses dynamic... just... history)